### PR TITLE
Include missing XMLBeans schemas in the native image

### LINKED ISSUE #59: Native-Image Schema is missing: org.apache.poi.schemas.ooxml.system.ooxml.shapelayoutelement
Hey, 

first, thanks for your amazing work! 

Currently I try to upgrade to quarkus 3 with quarkiverse.tika 2.2 and quarkiverse.poi 2.2. 

Until now I used quarkus 2, quarkiverse.tike 1.X and poi (but not as quarkus extension). Thats why I had to used [native-image tracing agent](https://www.graalvm.org/latest/reference-manual/native-image/metadata/AutomaticMetadataCollection/) to getter native-image configs. Hope this extension will now make this obsolete. 

I process mainly excel files. In JVM mode all my testfiles works with this extension. In native mode unfortunately one file couldn't process and throws this error: 

```sh
Caused by: org.apache.xmlbeans.SchemaTypeLoaderException: XML-BEANS compiled schema: Could not locate compiled schema resource org/apache/poi/schemas/ooxml/system/ooxml/shapelayoutelement.xsb (org.apache.poi.schemas.ooxml.system.ooxml.shapelayoutelement) - code 0
        at org.apache.xmlbeans.impl.schema.XsbReader.<init>(XsbReader.java:63)
        at org.apache.xmlbeans.impl.schema.SchemaTypeSystemImpl.resolveHandle(SchemaTypeSystemImpl.java:931)
        at org.apache.xmlbeans.SchemaComponent$Ref.getComponent(SchemaComponent.java:111)
        at org.apache.xmlbeans.SchemaGlobalElement$Ref.get(SchemaGlobalElement.java:74)
        at org.apache.xmlbeans.impl.schema.SchemaTypeLoaderBase.findElement(SchemaTypeLoaderBase.java:103)
        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.createElementType(SchemaTypeImpl.java:993)
        at org.apache.xmlbeans.impl.values.XmlObjectBase.create_element_user(XmlObjectBase.java:879)
        at org.apache.xmlbeans.impl.store.Xobj.getUser(Xobj.java:1597)
        at org.apache.xmlbeans.impl.store.Cur.getUser(Cur.java:2571)
        at org.apache.xmlbeans.impl.store.Cur.getObject(Cur.java:2565)
        at org.apache.xmlbeans.impl.store.Cursor._getObject(Cursor.java:814)
        at org.apache.xmlbeans.impl.store.Cursor.syncWrapHelper(Cursor.java:2529)
        at org.apache.xmlbeans.impl.store.Cursor.syncWrap(Cursor.java:2460)
        at org.apache.xmlbeans.impl.store.Cursor.getObject(Cursor.java:2086)
        at org.apache.poi.xssf.usermodel.XSSFVMLDrawing.read(XSSFVMLDrawing.java:155)
        at org.apache.poi.xssf.usermodel.XSSFVMLDrawing.<init>(XSSFVMLDrawing.java:123)
        at org.apache.poi.ooxml.POIXMLFactory.createDocumentPart(POIXMLFactory.java:61)
        at org.apache.poi.ooxml.POIXMLDocumentPart.read(POIXMLDocumentPart.java:661)
        at org.apache.poi.ooxml.POIXMLDocumentPart.read(POIXMLDocumentPart.java:678)
        at org.apache.poi.ooxml.POIXMLDocument.load(POIXMLDocument.java:165)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:260)
        at org.apache.poi.xssf.usermodel.XSSFWorkbookFactory.createWorkbook(XSSFWorkbookFactory.java:118)
        at org.apache.poi.xssf.usermodel.XSSFWorkbookFactory.create(XSSFWorkbookFactory.java:98)
        at org.apache.poi.xssf.usermodel.XSSFWorkbookFactory.create(XSSFWorkbookFactory.java:36)
        at org.apache.poi.ss.usermodel.WorkbookFactory.lambda$create$2(WorkbookFactory.java:224)
        at org.apache.poi.ss.usermodel.WorkbookFactory.wp(WorkbookFactory.java:329)
        at org.apache.poi.ss.usermodel.WorkbookFactory.create(WorkbookFactory.java:224)
        at org.apache.poi.ss.usermodel.WorkbookFactory.create(WorkbookFactory.java:185)
        at io.quarkiverse.poi.it.POIResource.specialFile(POIResource.java:123)
        at io.quarkiverse.poi.it.POIResource_ClientProxy.specialFile(Unknown Source)
        at java.base@17.0.6/java.lang.reflect.Method.invoke(Method.java:568)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
        ... 17 more
``` 

As described here: https://poi.apache.org/help/faq.html XMLBeans should be compiled and available through poi-ooxml-full jar

> To use the new OOXML file formats, POI requires a jar containing the file format XSDs, as compiled by [XMLBeans](https://xmlbeans.apache.org/). These XSDs, once compiled into Java classes, live in the org.openxmlformats.schemas namespace.

I don't know if this is the right (or the right schema), but when I debug my [test](https://github.com/knuspertante/quarkus-poi/blob/b13beddf3c9277b1c5b885e9f4819567fa6e8487/integration-tests/src/test/java/io/quarkiverse/poi/it/POIResourceTest.java#LL34C1-L37C6) (could be used also as reproducer), the following schemas are available in [POIProcessor#registerXMLBeansClassesForReflection](https://github.com/quarkiverse/quarkus-poi/blob/9f3fae8ceea7c7ead7c1c6de2f201093252144af/deployment/src/main/java/io/quarkiverse/poi/deployment/POIProcessor.java#L51-L56C22)
- com.microsoft.schemas.office.office.impl.CTShapeLayoutImpl
- com.microsoft.schemas.office.office.impl.ShapelayoutDocumentImpl 

Reproduce: 

1. clone https://github.com/knuspertante/quarkus-poi/tree/bug/shapelayoutelement
2. mvn clean verify -Pnative-image

Thanks for your help!

### PR DESCRIPTION
**Pull request by knuspertante**
- Fixes #59

### CODE REVIEW

**Comment by melloware** [comment]
Build is running now.

**Comment by gastaldi** [comment]
Can you format the sources using `mvn clean process-sources`?

**Comment by gastaldi** [comment]
@all-contributors add @knuspertante for bug and test

**Comment by gastaldi** [comment]
@knuspertante I pushed the fix to your branch and renamed the PR (hope you don't mind) 😄 
Can you give it a shot?

Also I'd appreciate the feedback from the @quarkiverse/quarkiverse-poi team 😉

**Review by melloware — APPROVED** [review]
Nice fix!

File: deployment/src/main/java/io/quarkiverse/poi/deployment/POIProcessor.java [hunk 1]
```diff
@@ -70,11 +70,9 @@ public ReflectiveClassBuildItem registerLog4jClassesForReflection() {
...
 
     @BuildStep
-    public NativeImageResourcePatternsBuildItem registerTextResources() {
+    public NativeImageResourcePatternsBuildItem registerResources() {
         return new NativeImageResourcePatternsBuildItem.Builder().includePatterns(
-                "org/apache/poi/ss/formula/function/.*\\.txt"
-
-        ).build();
+                "org/apache/poi/ss/formula/function/.*\\.txt",
+                "org/apache/poi/schemas/ooxml/system/ooxml/.*\\.xsb").build();
```
Comment (gastaldi) [inline deployment/src/main/java/io/quarkiverse/poi/deployment/POIProcessor.java:76]: Perhaps we should add all *.xsb under the `org/apache/poi/schemas/ooxml/` directory, WDYT @melloware ?

```suggestion
                "org/apache/poi/schemas/ooxml/.*\\.xsb").build();
```

**Review by gastaldi — APPROVED** [review]

**Comment by knuspertante** [comment]
@gastaldi thanks! LGTM and works very well!

That's funny, because the missing resource was also my first thought but after i monitored the jvm mode with the tracing agent, this entry was not included in the resource-config.json. 
That's why I thought it can't be that and the xsb are already compiled as a class.

Thanks for adding me as a contributor, I appreciate that!

@gastaldi @melloware I think this could be merged or do I have to do something, like rename the test / file, clean up the commit history, ... ?

**Comment by melloware** [comment]
I squashed it for you!

**Comment by gastaldi** [comment]
@melloware fancy backporting this to 1.x and releasing both?

**Comment by melloware** [comment]
yep i will put it on my list today!

**Pull request merged**